### PR TITLE
Project States

### PR DESCRIPTION
**Pull request by sammachin**
Initial outline of project states for discussion between now and Friday 18th

### CODE REVIEW

File: development/index.md [hunk 1]
```diff
@@ -75,3 +75,19 @@ or stories will be completed.
...
+
+### Project Board
+The project board holds all issues and PR's across all our repos.
+
+Within the project board we use the following status's and move between them as shown.
+
+**No Status** This is the default for all new issues, The Product Owner will review them and move to an appropriate next status.
+**Backlog** This is for all work that we know needs to be done eventually but has not yet been prioritized to a milestone.
+**In Design** Items for this or the next milestone which need active planning or design consideration before work can start
+**ToDo**  Ready to start development work, person may or may not be assigned
```
Comment (joepavitt) [inline development/index.md:88]: - worth emphasising here a need for a milestone to be attached, given that the "Backlog" definition is that hasn't been assigned to a milestone.

File: development/index.md [hunk 2]
```diff
@@ -75,3 +75,19 @@ or stories will be completed.
...
+
+
+### Project Board
+The project board holds all issues and PR's across all our repos.
+
+Within the project board we use the following status's and move between them as shown.
+
+**No Status** This is the default for all new issues, The Product Owner will review them and move to an appropriate next status.
+**Backlog** This is for all work that we know needs to be done eventually but has not yet been prioritized to a milestone.
+**In Design** Items for this or the next milestone which need active planning or design consideration before work can start
```
Comment (joepavitt) [inline development/index.md:87]: "before development can start" - I'd like to think design & planning still count as work :)

**Comment by knolleary** [comment]
We need states that are useful for the different audiences of the project board and the phases an item can go through.

I may be overthinking it, but as we grow the team, there will be more of a separation between design and engineering - specifying the 'what' a feature requires ( which can cover business goals, UX, External API, Architecture, Database models), and then the implementation of that feature.

We also have to recognise there are different axis here - the state of the item, the milestone it is assigned to etc.

---
*Edit* - having finished writing this, I'm coming back to add a couple observations.

First, my suggestions here are pretty aligned to what is in the PR. The key thing is I've tried to spell out a bit more on what some of the states mean - and what work happens when. I've also renamed a couple - notably 'ToDo' -> 'Ready'.

Second, I've named roles of people in a few places. That's to make clear who has the responsibility for a task - but that doesn't mean they have to do it in isolation. For example, the prioritisation of the backlog has to be a collaboration between Product and Engineering - and anyone should be able to make the case for a particular item to be brought forward.

---

So, here are the states I think we could have - at this time:


> **Backlog** : all items not currently assigned to a milestone. It contains Product Items (Epics/Stories) that are prioritised by the Product Owner and Technical Chores (Tasks) that are prioritised by the CTO.

To properly plan an item it needs an estimated size. Sizing items in agile is a whole topic in its own right and something we need to discuss as a team. But the point is, that's a piece of work that has to happen as part of the planning work. That should also identify what prerequisites the item has. The Product Owner/CTO ought to be regularly reviewing the backlog and identifying items that are likely to come up. They can then coordinate sizing work for those items.

From there, the item moves to...

> **Design** : work on the item is focused on design work. Creating wireframe mockups, sketching public API changes and Database model requirements.

We aren't yet up to full speed on this side of things yet - but our goal should be that we can use figma to quickly mock up any UX changes. That work should be doable by anyone in the team, but with review/input by Design/CTO/Product. We risk our current Design team of 1 person becoming a bottle neck here - so we need to be pragmatic whilst also moving to a design-first approach.

To support this, I propose we make more formal use of issue labels. This can also help raise awareness of the impact an item has. For example, the following labels could be used to indicate the expected areas it will touch: (which I hope are self-explanatory)
 - `frontend`
 - `database`
 - `API`

(And I'm sure there will be others that emerge)

Once an item has sufficient design work to proceed (ie a figma mockup, a description of the intended behaviour) it can move to:

> **Ready** : this is the engineering backlog for a milestone. Items in this state should have enough information in them for anyone to pick and start implementing what is described.

> **In Progress** : items that are assigned to someone and they are working on it

> **Review** : items that are ready to be reviewed. 

> **Done** : finished items. But we do need a clear definition of 'done'.

**Comment by joepavitt** [comment]
Agree entirely with everything you've written @knolleary

**Comment by ZJvandeWeg** [comment]
Closing this PR as #57 has been merged.

**Pull request closed without merging**